### DISCUSSION TITLE
State setter is of type undefined

### DISCUSSION THREAD

**Discussion by isitreallyalive** [pmndrs/zustand#839] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/19301186/156259426-078252b1-156c-48be-ac8d-18cce7a69ea5.png)

I installed zustand and pasted the example in the readme into my file, but my editor is telling me that the state setter is of type unknown. As you can see, there are two unknowns in the generic for the method.

![image](https://user-images.githubusercontent.com/19301186/156259567-70fd6a0b-fa5e-40e6-8da3-c22173812241.png)

Does anyone have any idea what the issue could be?

**Comment by dai-shi** [comment, marked as answer]
If you are using JS without types, but your editor complains, you want to disable type checking.

If you are okay to use TS, https://github.com/pmndrs/zustand#typing-your-store-and-combine-middleware may help.

**Reply by asktree** [reply]
> If you are okay to use TS, https://github.com/pmndrs/zustand#typing-your-store-and-combine-middleware may help.

I'm confused -- can't we type `create` such that it simply infers the type of our state, without needing us to write a type for the state and pass it in?

**Reply by dai-shi** [reply]
The best I came up with is the `combine` middleware. Is your question how it's impossible without it?

**Reply by asktree** [reply]
@dai-shi yeah thats probably my question

**Reply by dai-shi** [reply]
To my knowledge, it's impossible due to TS limitation.
I'd rather suggest to go with this new pattern: https://github.com/pmndrs/zustand/wiki/Practice-with-no-store-actions

**Reply by asktree** [reply]
That pattern is neat, thanks.
Sad to hear that there's a limitation here. If I have time next month I will tinker with it and see if I can come up with something.
Thanks for your hard work! Zustand is great.
